### ISSUE TITLE
Tidy up `summary` and `as_gt` functions

### ISSUE THREAD

**Issue by LittleBeannie** [Merck/gsDesign2#282]
The summary and as_gt functions in their current state are extremely lengthy. A preferable approach would be to condense them into a shorter function that utilizes sub-functions where necessary.

**Comment by LittleBeannie** [comment]
Hi @jdblischak , considering we have couple of PRs to improve `as_gt` and `as_rtf` (such as https://github.com/Merck/gsDesign2/pull/458, https://github.com/Merck/gsDesign2/pull/450, https://github.com/Merck/gsDesign2/pull/449), can we close this issue? I can create another issue to tidy-up the nested if-else in `summary.gs_design` in another issue, to make the issue more specific.

**Comment by yihui** [comment]
Or I can create a PR to clean up the summary functions, and close this issue once that PR is merged.
